File: CascadingComboboBox/BankCB.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Windows.Forms.VisualStyles.VisualStyleElement.TreeView;

namespace CascadingComboboBox
{
    internal class BankCB
    {
        public int BankID { get; set; }
        public string BankName { get; set; }
        public BindingList<BranchCB> Branches { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine("----------------------------------------------------------");
            sb.AppendLine("BankID: " + BankID + " Name: " + BankName + " Branches:...");
            if (Branches.Count > 1)
            {
                foreach (BranchCB branch in Branches)
                {
                    if (branch.BranchID != 0)
                    {
                        sb.AppendLine(branch.ToString());
                    }
                }
            }
            else
            {
                sb.AppendLine("No Branches");
            }
            return sb.ToString();
        }
    }
}

File: CascadingComboboBox/BankCB.vb
Imports System.ComponentModel
Imports System.Text
Imports System.Windows.Forms.VisualStyles.VisualStyleElement.TreeView

Namespace CascadingComboboBox
    Friend Class BankCB
        Public Property BankID As Integer
        Public Property BankName As String
        Public Property Branches As BindingList(Of BranchCB)

        Public Overrides Function ToString() As String
            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendLine("----------------------------------------------------------")
            sb.AppendLine("BankID: " & BankID.ToString() & " Name: " & BankName & " Branches:...")
            If Branches.Count > 1 Then
                For Each branch In Branches
                    If branch.BranchID <> 0 Then
                        sb.AppendLine(branch.ToString())
                    End If
                Next
            Else
                sb.AppendLine("No Branches")
            End If
            Return sb.ToString()
        End Function
    End Class
End Namespace
